server.go

```diff
@@ -7,21 +7,51 @@ import (
 	"github.com/gorilla/mux"
 	"log"
 	"net/http"
+	"html/template"
+	"path"
+	
+
 )
 
-func main() {
 
+func HomePage(w http.ResponseWriter, r *http.Request){
```

Comment:
For future reference, make each pull request one self-contained change.  This one could be split into fixing the user model, and adding the homepage route.